bigquery_etl/cli/utils.py

```diff
@@ -104,6 +105,25 @@ def paths_matching_name_pattern(pattern, sql_path, project_id, files=("*.sql")):
     callback=is_valid_dir,
 )
 
+
+def generate_tmp_dir(ctx, param, value):
+    """Generate temporary directory if no default directory is provided."""
+    if value is None:
+        tmp_dir = Path(tempfile.mkdtemp()) / "sql"
```

Comment:
Do we ever clean up this directory? If we're generating a random tmp dir, I'd expect this code to also handle cleaning up the directory when it's done.

Reply:
Can we use the pattern described in https://click.palletsprojects.com/en/8.0.x/advanced/#managing-resources ?

Reply:
Interesting, yes! It seems like the context object cannot be renamed, e.g. I was hoping `ctx.tmp_dir` would be allowed, but only `ctx.obj` works.

Reply:
Ah actually `ctx.obj["TMP_DIR"]` might be a good alternative here